klippy/extras/bed_mesh.py

```diff
@@ -2,10 +2,14 @@
 #
 # Copyright (C) 2018  Kevin O'Connor <kevin@koconnor.net>
 # Copyright (C) 2018-2019 Eric Callahan <arksine.code@gmail.com>
+# Copyright (C) 2023  Kyle Hansen (Kyleisah) <kyleisah@gmail.com>
+# Copyright (C) 2023  Brandon Nance <bwnance@gmail.com>
+# Copyright (C) 2023  Mitko Haralanov <voidtrance@gmail.com>
```

Comment:
Generally we only add a copyright on initial submissions.  We'd like to put everyone in there who has contributed, however we want to keep header comments from growing out of control.

Reply:
So how are attributions to the codebase by other people handled? IANAL, but my understanding is that under the GPL, code authors retain their copyright for the code that they've submitted. In such cases, shouldn't the copyright notice be added?

Reply:
I believe the copyright is implied when the DCO is accepted in the commit.  This is covered in point 3 of [what to expect in a review](https://www.klipper3d.org/CONTRIBUTING.html#what-to-expect-in-a-review):

```
New C files and Python files should have an unambiguous copyright statement.
See the existing files for the preferred format. Declaring a copyright on an existing
file when making minor changes to that file is discouraged.
```

This could be considered more than a minor change, so perhaps keeping the copyright is ok.  I'll let Kevin chime in on that.

Reply:
OK, the latest revision still contains the copyrights. I will adjust based on Kevin's input.

Reply:
We don't want the copyright header to turn into a "laundry list of everyone that has worked on the code".  Use the commit messages to document who worked on the code - preferably in the form of a signed-off-by line.  In my experience, if the copyright block does become a "laundry list" it tends to grow out of control - requiring scrolling through pages just to get to the code - which becomes a maintenance burden.

When I look at bed_mesh.py it seems to overwhelmingly be Eric's code.  If there is a feeling that a major contribution is being made then it's fine to add additional copyright(s).  I'd say that determination is up to Eric.  For what it is worth, at a quick glance, adding three copyright statements for 100 lines of code seems excessive to me.

FWIW, you can remove my copyright from bed_mesh.py (Eric added my copyright back in the original bed_mesh.py commit - it is not necessary).

Hope that helps.

-Kevin

Reply:
Just chiming in here regarding the copyright, I am fine with my name being removed as it was VoidTrance who did 99% of the work implementing the logic of KAMP (my project, the repo is in my GitHub profile). That said, some credit somewhere in the klipper repo would be favorable of course. I approached VoidTrance with the idea and helped any way I could, but I cannot take credit for the work he’s done here. I laid some foundation, VoidTrance built the house. 

As for Brandon Nance, he tried to help me for a couple hours when we initially had the idea to bring this into mainline klipper, we wrote a few lines of code together, but easily 90+% of the work was done by VoidTrance. 

I don’t want to complicate things with who did what, just want to give Kevin some backstory so he can make the final decision without missing any nuance here.

Reply:
Thanks for the input Kevin.

Eric, I guess it's your call. Getting this in is more important to me than having my copyright in the file so I am fine either way.

Reply:
> I am fine with my name being removed 

> Getting this in is more important to me than having my copyright in the file so I am fine either way.

Thanks.  I think we should not add these new copyright statements then.  The more brief we can keep the header the more brief we'll be able to keep it in the future.

If you're in that area of the code, feel free to remove my copyright statement.

Thanks again.

-Kevin

Reply:
Done.